mutiny-core/src/lib.rs

```diff
@@ -247,54 +246,128 @@ pub struct MutinyWalletConfig {
+            skip_device_lock: false,
             safe_mode: false,
-            skip_hodl_invoices,
+            skip_hodl_invoices: true,
         }
     }
 
-    pub fn with_do_not_connect_peers(mut self) -> Self {
+    #[cfg(target_arch = "wasm32")]
+    pub fn with_websocket_proxy_addr(&mut self, websocket_proxy_addr: String) {
+        self.websocket_proxy_addr = Some(websocket_proxy_addr);
+    }
+
+    pub fn with_network(&mut self, network: Network) {
+        self.network = network;
+    }
+
+    pub fn with_user_esplora_url(&mut self, user_esplora_url: String) {
+        self.user_esplora_url = Some(user_esplora_url);
+    }
+
+    pub fn with_user_rgs_url(&mut self, user_rgs_url: String) {
+        self.user_rgs_url = Some(user_rgs_url);
+    }
+
+    pub fn with_lsp_url(&mut self, lsp_url: String) {
+        self.lsp_url = Some(lsp_url);
+    }
+
+    pub fn with_lsp_connection_string(&mut self, lsp_connection_string: String) {
+        self.lsp_connection_string = Some(lsp_connection_string);
+    }
+
+    pub fn with_lsp_token(&mut self, lsp_token: String) {
+        self.lsp_token = Some(lsp_token);
+    }
+
+    pub fn with_auth_client(&mut self, auth_client: Arc<MutinyAuthClient>) {
+        self.auth_client = Some(auth_client);
+    }
+
+    pub fn with_subscription_url(&mut self, subscription_url: String) {
+        self.subscription_url = Some(subscription_url);
+    }
+
+    pub fn with_scorer_url(&mut self, scorer_url: String) {
+        self.scorer_url = Some(scorer_url);
+    }
+
+    pub fn do_not_connect_peers(&mut self) {
         self.do_not_connect_peers = true;
-        self
     }
 
-    pub fn with_safe_mode(mut self) -> Self {
+    pub fn with_skip_device_lock(&mut self) {
+        self.skip_device_lock = true;
+    }
+
+    pub fn with_safe_mode(&mut self) {
         self.safe_mode = true;
-        self.with_do_not_connect_peers()
+        self.skip_device_lock = true;
+    }
+
+    pub fn do_not_skip_hodl_invoices(&mut self) {
+        self.skip_hodl_invoices = false;
+    }
+
+    pub fn build(self) -> MutinyWalletConfig {
+        MutinyWalletConfig {
+            xprivkey: self.xprivkey,
+            #[cfg(target_arch = "wasm32")]
+            websocket_proxy_addr: self.websocket_proxy_addr,
+            network: self.network,
+            user_esplora_url: self.user_esplora_url,
+            user_rgs_url: self.user_rgs_url,
+            lsp_url: self.lsp_url,
+            lsp_connection_string: self.lsp_connection_string,
+            lsp_token: self.lsp_token,
+            auth_client: self.auth_client,
+            subscription_url: self.subscription_url,
+            scorer_url: self.scorer_url,
+            do_not_connect_peers: self.do_not_connect_peers,
+            skip_device_lock: self.skip_device_lock,
+            safe_mode: self.safe_mode,
+            skip_hodl_invoices: self.skip_hodl_invoices,
+        }
     }
 }
 
+#[derive(Clone)]
+pub struct MutinyWalletConfig {
```

Comment:
since we want people to use the builder this should be `pub(crate)`

Reply:
The builder returns `MutinyWalletConfig`, so if we have it behind `crate`, it would be unable to be used. Having the members non `pub` should eliminate the ability to init `MutinyWalletConfig` directly.